File: sdk/src/main/java/com/cxense/cxensesdk/model/UserDataRequest.java
package com.cxense.cxensesdk.model;

import android.support.annotation.NonNull;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

/**
 * Request object for getting user from server
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
@SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
public final class UserDataRequest extends UserIdentity {
    @JsonProperty("groups")
    List<String> groups;
    @JsonProperty("recent")
    Boolean isRecent;
    @JsonProperty("identityTypes")
    List<String> identityTypes;

    public UserDataRequest(@NonNull UserIdentity identity, List<String> groups, Boolean recent, List<String> identityTypes) {
        super(identity);
        isRecent = recent;
        if (groups != null)
            this.groups = new ArrayList<>(groups);
        if (identityTypes != null)
            this.identityTypes = new ArrayList<>(identityTypes);
    }
}

File: sdk/src/main/java/io/piano/android/cxense/model/UserDataRequest.kt
package io.piano.android.cxense.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Request object for getting user from server
 *
 */
@JsonClass(generateAdapter = true)
public class UserDataRequest(
    type: String,
    id: String,
    @Json(name = "groups") public val groups: List<String>? = null,
    @Json(name = "recent") public val recent: Boolean? = null,
    @Json(name = "identityTypes") public val identityTypes: List<String>? = null,
) : UserIdentity(type, id) {
    public constructor(
        userIdentity: UserIdentity,
        @Json(name = "groups") groups: List<String>? = null,
        @Json(name = "recent") recent: Boolean? = null,
        @Json(name = "identityTypes") identityTypes: List<String>? = null,
    ) : this(userIdentity.type, userIdentity.id, groups, recent, identityTypes)
}
